### ISSUE TITLE
long option aliases

### ISSUE THREAD

**Issue by rbdavis** [PaulWessel/CodingTasks#5]
Hi Paul,

I've now coded and done some initial testing on long-option aliases. Aliases can now be spec'd  for both (i) the _--whatever_ long-option trigger flags as well as (ii) the long-form _+whatever_ modifier and directive strings.

Before I do some more thorough testing and make a PR I was wondering if you'd like me to code any particular long-option aliases for the common options or even for any module-specific options. Two candidates that come immediately to mind would be **--reg** and **--proj** in the common options. Otherwise, none of the current long-options tables include any aliases at the moment. (I did initial testing with a couple of temporary aliases that are now removed.)

**Comment by PaulWessel** [comment]
I guess we should look to see what aliases Joaquim is using in Julia for the common options.  Sorry, but until the overs come on Friday I have no time to do anything related to coding... But --proj is fine and probably --reg as well.  I was thinking --domain as well even though not starting with r.

**Comment by rbdavis** [comment]
OK, I will go back through all the modules I've worked on, look at the Julia strings, and add aliases where it seems appropriate. Doing that and re-running testing should keep me busy for a bit. Hope the packing up goes OK!

**Comment by rbdavis** [comment]
The Julia package has a lot of aliases which I'm not sure are essential to include in the C longopts. For  instance, **blockmean -A** can be spec'd with either **field** or **fields**. The C table currently uses **fields**.

I think (to avoid not particulary helpful clutter) until you tell me otherwise I will take the approach of including at least one, but not necessarily more than one, of the Julia aliases in the C table. In this case, for instance, I will leave the blockmean C table alone. LMK if you want otherwise.

**Comment by PaulWessel** [comment]
Yes, I would not go overboard on that.  I can understand he thought field is useful for one item and fields for more than one but that is getting into the weeds.  Since the option can take more than one field it should be fields.

**Comment by rbdavis** [comment]
Here's the complete list of alias additions I made yesterday, and will begin testing shortly. Please advise any changes and I will correct. Each line below shows the short-option flag and the full list of long-option names associated (with the 'official' long-option name first and the aliases following). I took the liberty of adding one modifier alias ( +rect ) to the common long options, otherwise none anywhere else.

Note that Joaquim uses **crossection** for **grdinterpolate -E**, which I have copied unchanged but do not like as it's a misspelling. It should be **cross-section**, but we shouldn't use a hyphen and **crosssection** looks weird. My suggestion would be **xsection**, but this does not provide compatibility with what he's done.

gmt_common_longoptions.h
        P projection|proj
        R region|reg|domain
          +r rectangular|rect

longopt/grdimage_inc.h
        C cpt|cmap
        Q transvalue|alphacolor

longopt/grdinterpolate_inc.h
        E profile|crossection
        T timeinc|inc|range

longopt/grdtrack_inc.h
        D linefile|dfile

longopt/mapproject_inc.h
        W mapinfo|mapsize

longopt/pscoast_inc.h
        C lakes|riverfill

longopt/pswiggle_inc.h
        T trackpen|track
        W outlinepen|pen

longopt/psxy_inc.h
        C cpt|cmap
        Z zvalue|level

longopt/surface_inc.h
        M maskradius|mask

**Comment by rbdavis** [comment]
Just changed my mind about **crossection** vs. **crosssection** in **grdinterpolate**, opting for **crosssection**. I am already using the latter in the seis/pscoupe module, probably best to be consistent. (Although I guess I'd prefer **xsection** for both.) 

Done with all testing now, will start cleaning up to submit PR soon.
